src/app/core/cache/object-cache.reducer.ts

```diff
@@ -0,0 +1,59 @@
+import { ObjectCacheAction, ObjectCacheActionTypes, AddToObjectCacheAction, RemoveFromObjectCacheAction } from "./object-cache.actions";
+import { hasValue } from "../../shared/empty.util";
+import { CacheEntry } from "./cache-entry";
+
+export interface CacheableObject {
+  uuid: string;
+}
+
+export class ObjectCacheEntry implements CacheEntry {
+  data: CacheableObject;
+  timeAdded: number;
+  msToLive: number;
+}
+
+export interface ObjectCacheState {
+  [uuid: string]: ObjectCacheEntry
+}
+
+// Object.create(null) ensures the object has no default js properties (e.g. `__proto__`)
+const initialState: ObjectCacheState = Object.create(null);
+
+export const objectCacheReducer = (state = initialState, action: ObjectCacheAction): ObjectCacheState => {
+  switch (action.type) {
+
+    case ObjectCacheActionTypes.ADD: {
+      return addToObjectCache(state, <AddToObjectCacheAction>action);
+    }
+
+    case ObjectCacheActionTypes.REMOVE: {
+      return removeFromObjectCache(state, <RemoveFromObjectCacheAction>action)
+    }
+
+    default: {
+      return state;
+    }
+  }
+};
+
+function addToObjectCache(state: ObjectCacheState, action: AddToObjectCacheAction): ObjectCacheState {
+  return Object.assign({}, state, {
+    [action.payload.objectToCache.uuid]: {
```

Comment:
I get this error when asking for an item id that can't be found: `TypeError: Cannot read property 'uuid' of undefined`
Note: I'm still testing using the HomeComponent for now. So I had to comment out the "doRehydrate" in "browser.module.ts" first, because the state from the home component in preloaded by the "universal cache". The errors don't show up on the server side.